docs/plugins/outputs/kafka.asciidoc

```diff
@@ -20,11 +21,24 @@ include::{include_path}/plugin_header.asciidoc[]
 
 ==== Description
 
-Write events to a Kafka topic.
+Write events to a Kafka topic. This uses the Kafka Producer API to write messages to a topic on
+the broker.
 
-This plugin uses Kafka Client 0.10.0.1. For broker compatibility, see the official https://cwiki.apache.org/confluence/display/KAFKA/Compatibility+Matrix[Kafka compatibility reference].
+Here's a compatibility matrix that shows the Kafka client versions that are compatible with each combination
+of Logstash and the Kafka output plugin: 
 
-If you're using a plugin version that was released after {version}, see the https://www.elastic.co/guide/en/logstash/master/plugins-inputs-kafka.html[latest plugin documentation] for updated information about Kafka compatibility. If you require features not yet available in this plugin (including client version upgrades), please file an issue with details about what you need.
+[options="header"]
+|==========================================================
+|Kafka Client Version |Logstash Version |Plugin Version |Why?
```

Comment:
@jsvd Same comment/question applies here about whether we want this matrix in 5.6.